back-office/src/user/actions.js

```diff
@@ -0,0 +1,81 @@
+//@flow
+import type {
+  AddUserAction,
+  UpdateCurrentUserPropertyAction,
+  UpdateUserAction,
+  ResetCurrentUserAction,
+} from './type'
+
+import { userBackend } from '../api'
+
+export const actions = {
+  ADD_USER_SUCCESS: "USER/ADD_USER_SUCCESS",
+  UPDATE_CURRENT_USER_PROPERTY: "USER/UPDATE_CURRENT_USER_PROPERTY",
+  UPDATE_USER_SUCCESS: "USER/UPDATE_USER_SUCCESS",
+  RESET_CURRENT_USER: "USER/RESET_CURRENT_USER",
+  EDIT_USER: "USER/EDIT_USER",
+  DELETE_USER_SUCCESS: "USER/DELETE_USER_SUCCESS"
+}
+
+export const addUserSuccess = (user: User): AddUserAction => ({
+  type: actions.ADD_USER_SUCCESS,
+  payload: {
+    user: user
+  }
+})
+
+export const addUser = (user: User) => (dispatch) => {
+  userBackend.addUser(user)
+    .then((userAdded) => dispatch(addUserSuccess(userAdded)))
+    .then(() => dispatch(resetCurrentUser()))
+    .catch(error => {
+      console.error(error)
```

Comment:
You can use `toastr.error(message)` to display an error on the screen. You can then just rethrow the exception to let the promise fail. Uncaught promises are already shown on the console.